### PR TITLE
Webpack: Add hot reloading

### PR DESCRIPTION
**Pull request by sorja**
Adds hot reloading for frontend, caching for server (faster reloads/builds : 6s => 2s)
Note: Also adds morgan for server side request logging (on dev, local)

### CODE REVIEW

File: webpack.config.babel.js [hunk 1]
```diff
@@ -67,7 +72,7 @@ const plugins = [
     filename: fontCssFileName,
   }),
   new CleanUpStatsPlugin(),
-]
+].filter(Boolean)
```
Comment (SteRiccio) [inline webpack.config.babel.js:75]: it's not very clear... can you push the development plugins in a "if" block instead?

File: server/system/appCluster.js [hunk 2]
```diff
@@ -31,6 +33,10 @@ export const run = async () => {
 
   const app = express()
 
+  if (ProcessUtils.ENV.nodeEnv === 'development') {
```
Comment (SteRiccio) [inline server/system/appCluster.js:36]: you can use ProcessUtils.isEnvDevelopment instead

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: webpack.config.babel.js [hunk 3]
```diff
@@ -93,17 +101,50 @@ const webPackConfig = {
...
+            loader: 'babel-loader',
+            options: {
+              plugins: isDevelopment ? [require.resolve('react-refresh/babel')] : [],
+            },
+          },
+        ],
       },
       {
-        test: /\.(sa|sc|c)ss$/,
+        test: /\.(sass|scss|css)$/,
```
Comment (minotogna) [inline webpack.config.babel.js:147]: was this change necessary ?

Comment (SteRiccio) [reply webpack.config.babel.js:147]: I don't know if it's necessary or not but to me it looks more clear in this way

Comment (ramirobg94) [reply webpack.config.babel.js:147]: For me too 👍

**Review by ramirobg94 — APPROVED** [review]

File: webpack.config.babel.js [hunk 4]
```diff
@@ -93,17 +101,50 @@ const webPackConfig = {
...
     publicPath: '/',
   },
+  devServer: {
+    hot: true,
+    index: '',
+    proxy: [
+      {
+        // Proxy all server-served routes:
+        context: ['/img', '/api', '/auth', '/socket.io', 'sockjs-node'],
+        target: 'http://localhost:9090',
```
Comment (ramirobg94) [inline webpack.config.babel.js:111]: should this be moved to some var?

File: webpack.config.babel.js (hunk 3 above)
Comment (sorja) [reply webpack.config.babel.js:147]: For grepping

File: webpack.config.babel.js (hunk 4 above)
Comment (sorja) [reply webpack.config.babel.js:111]: the target?

Comment (sorja) [reply webpack.config.babel.js:111]: I think it's clear like this - it is not used twice

**Review by minotogna — APPROVED** [review]

**Comment by minotogna** [comment]
can we merge this branch?

**Review by SteRiccio — APPROVED** [review]
